addons/web/static/src/views/fields/monetary/monetary_field.scss

```diff
@@ -1,3 +1,6 @@
 .o_field_widget.o_field_monetary input {
     width: 100px;
 }
+.o_field_widget.o_field_monetary_responsive input{
+    width: 100% !important;
```

Comment:
Meh, why important ? Can't you make better rules with either the "responsive" class or not being present ?